File: OpenEdXMobile/src/main/java/org/edx/mobile/view/adapters/DiscussionTopicsAdapter.java
package org.edx.mobile.view.adapters;

import android.content.Context;
import androidx.core.view.ViewCompat;
import android.view.View;
import android.widget.AdapterView;
import android.widget.TextView;

import com.google.inject.Inject;

import org.edx.mobile.R;
import org.edx.mobile.core.IEdxEnvironment;
import org.edx.mobile.discussion.DiscussionTopicDepth;

public class DiscussionTopicsAdapter extends BaseListAdapter<DiscussionTopicDepth> {

    private final int childPadding;

    @Inject
    public DiscussionTopicsAdapter(Context context, IEdxEnvironment environment) {
        super(context, R.layout.row_discussion_topic, environment);
        childPadding = context.getResources().getDimensionPixelOffset(R.dimen.edx_margin);
    }

    @Override
    public void render(BaseViewHolder tag, DiscussionTopicDepth discussionTopic) {
        ViewHolder holder = (ViewHolder) tag;
        holder.discussionTopicNameTextView.setText(discussionTopic.getDiscussionTopic().getName());
        ViewCompat.setPaddingRelative(holder.discussionTopicNameTextView, childPadding * (1 + discussionTopic.getDepth()), childPadding, childPadding, childPadding);
    }

    @Override
    public BaseViewHolder getTag(View convertView) {
        return new ViewHolder(convertView);
    }

    @Override
    public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
    }

    private static class ViewHolder extends BaseViewHolder {
        final TextView discussionTopicNameTextView;

        private ViewHolder(View view) {
            this.discussionTopicNameTextView = (TextView) view.findViewById(R.id.discussion_topic_name_text_view);
        }
    }

}

File: OpenEdXMobile/src/main/java/org/edx/mobile/view/adapters/DiscussionTopicsAdapter.kt
package org.edx.mobile.view.adapters

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.core.view.ViewCompat
import androidx.recyclerview.widget.ListAdapter
import androidx.recyclerview.widget.RecyclerView
import org.edx.mobile.R
import org.edx.mobile.databinding.RowDiscussionTopicBinding
import org.edx.mobile.interfaces.OnItemClickListener
import org.edx.mobile.model.discussion.DiscussionTopicDepth
import org.edx.mobile.util.UiUtils

class DiscussionTopicsAdapter(
    private val listener: OnItemClickListener<DiscussionTopicDepth>
) : ListAdapter<DiscussionTopicDepth, DiscussionTopicsAdapter.DiscussionTopicViewHolder>(
    DiscussionTopicDepth.DiscussionTopicsComparator()
) {
    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): DiscussionTopicViewHolder {
        return DiscussionTopicViewHolder(
            RowDiscussionTopicBinding.inflate(
                LayoutInflater.from(
                    parent.context
                ), parent, false
            )
        )
    }

    override fun onBindViewHolder(
        holder: DiscussionTopicViewHolder,
        position: Int
    ) {
        val item = getItem(position)
        holder.binding.discussionTopicNameTextView.apply {
            text = item.discussionTopic.getTopicTitle(resources)

            if (getItemViewType(position) == VIEW_TYPE_HEADER) {
                UiUtils.setTextViewDrawableStart(
                    context, this, R.drawable.ic_star_rate,
                    R.dimen.edx_base, R.color.primaryBaseColor
                )
            }

            // Add padding based on thread's depth
            val padding = context.resources.getDimensionPixelOffset(R.dimen.edx_margin)
            ViewCompat.setPaddingRelative(
                this, padding * (1 + item.depth), padding, padding, padding
            )
            setOnClickListener { listener.onItemClick(item) }
        }
    }

    override fun getItemViewType(position: Int): Int {
        return if (position == 1) VIEW_TYPE_HEADER else VIEW_TYPE_ITEM
    }

    class DiscussionTopicViewHolder(val binding: RowDiscussionTopicBinding) :
        RecyclerView.ViewHolder(binding.root)

    companion object {
        private const val VIEW_TYPE_HEADER = 0
        private const val VIEW_TYPE_ITEM = 1
    }
}
